.github/workflows/codspeed.yml

```diff
@@ -0,0 +1,48 @@
+name: Benchmarks
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'next'
+    paths:
+      - 'packages/x-charts*/**'
+  pull_request:
+    types:
+      - labeled
+      - unlabeled
+      - opened
+      - synchronize
+      - reopened
+    branches:
+      - 'master'
+      - 'next'
+
+jobs:
+  benchmarks:
+    name: Benchmarks Charts
+    runs-on: ubuntu-latest
+    # Only run the benchmarks for pull requests with the label 'component: charts'
+    # Or for pushes to the master or next branch
+    if: "${{ github.event_name == 'push' ||  contains(github.event.pull_request.labels.*.name, 'component: charts') }}"
```

Comment:
Does it means that if I add a label "bug" and it already have a label "charts" the script will be run?

If yes, it might be easier to only do the `paths` check and listen to only to opened, synchronize, and reopened

Reply:
Yes, though we can change it so that the label event only triggers when we add the charts label.

Reply:
Not sure it's the best moment to provide feedback to developers.

Personally, adding the charts label is the first thing I do when opening a PR. And at that moment the code modification is rarely complete

Reply:
My goal was more for completeness. Since this is not a "required" workflow, we can't enforce it. If the user finishes the PR and `then` add the label, the performance check wouldn't run.

Ensuring it runs on `charts label added` and `pr changes when there is a charts label` ensures we always have this run on PR's it is expected to run

Reply:
> If the user finishes the PR and then add the label, the performance check wouldn't run.

Not if we replace the condition on labels by a condition on `paths`

But that's not a big deal. We can try and refine based on how it feels with usage